### ISSUE TITLE
Minimum Spanning Tree got illegal memory access error on ak2010 dateset

### ISSUE THREAD

**Issue by yzhwang** [gunrock/gunrock#24]
@ydwu I'm testing MST today and found out this bug.

```
bin/test_mst_7.0_x86_64 market /data/gunrock_dataset/large/ak2010/ak2010.mtx | tee result
Reading from /data/gunrock_dataset/large/ak2010/ak2010.mtx:
  Parsing MARKET COO format (45292 nodes, 108549 directed edges)... Done parsing (0s).
  Converting 45292 vertices, 217098 directed edges (unordered tuples) to CSR format...
Done converting (0s).
[../../gunrock/app/mst/mst_enactor.cuh, 511] MSTProblem cudaMemcpy vertex_flag to d_vertex_flag failed (CUDA error 77: an illegal memory access was encountered)
[test_mst.cu, 253] MST Problem Enact Failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 232] MSTProblem cudaMemcpy selector failed (CUDA error 77: an illegal memory access was encountered)
[test_mst.cu, 262] MST Problem Data Extraction Failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 143] GpuSlice cudaFree d_col_indices  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 146] GpuSlice cudaFree d_edge_weights failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 149] GpuSlice cudaFree d_reduced_vals failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 152] GpuSlice cudaFree  d_flags_array failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 155] GpuSlice cudaFree  d_keys_array  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 158] GpuSlice cudaFree d_temp_storage failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 161] GpuSlice cudaFree d_reduced_keys failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 164] GpuSlice cudaFree  d_successors  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 167] GpuSlice cudaFree d_row_offsets  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 170] GpuSlice cudaFree d_vertex_flag  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 173] GpuSlice cudaFree d_origin_nodes failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 176] GpuSlice cudaFree d_supervtx_ids failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 179] GpuSlice cudaFree d_origin_edges failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 182] GpuSlice cudaFree  d_mst_output  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 185] GpuSlice cudaFree  d_edge_flags  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 188] GpuSlice cudaFree d_tmp_storage  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 191] GpuSlice cudaFree d_super_edges  failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_problem.cuh, 195] GpuSlice cudaFree data_slices failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/problem_base.cuh, 126] GpuSlice cudaFree d_row_offsets failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/problem_base.cuh, 127] GpuSlice cudaFree d_column_indices failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/problem_base.cuh, 131] GpuSlice cudaFree frontier_queues.d_keys failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/problem_base.cuh, 132] GpuSlice cudaFree frontier_queues.d_values failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/problem_base.cuh, 131] GpuSlice cudaFree frontier_queues.d_keys failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/problem_base.cuh, 132] GpuSlice cudaFree frontier_queues.d_values failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_enactor.cuh, 144] MSTEnactor cudaFreeHost done failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/mst/mst_enactor.cuh, 148] MSTEnactor cudaEventDestroy throttle_event failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/enactor_base.cuh, 120] EnactorBase cudaFree d_node_locks failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/enactor_base.cuh, 121] EnactorBase cudaFree d_node_locks_out failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/util/kernel_runtime_stats.cuh, 143] KernelRuntimeStatsLifetime cudaFree d_stat failed:  (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/util/kernel_runtime_stats.cuh, 143] KernelRuntimeStatsLifetime cudaFree d_stat failed:  (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/util/cta_work_progress.cuh, 230] CtaWorkProgress cudaFree d_counters failed:  (CUDA error 77: an illegal memory access was encountered)

MINIMUM SPANNING TREE TEST
GPU - Computation Complete in 0.000000 msec.

REFERENCE TEST
CPU - Computation Complete in 138.863480 msec.
INCORRECT. 
CPU Computed Total Weight = 5414746115
GPU Computed Total Weight = 50497260931659
```

**Comment by ydwu** [comment]
So... MST only takes graphs with *#Connected Components == 1*. ak2010 has 50. So this is not a bug. I wish I can add some detection code for input graphs before running on GPU but don't know how. 

[yuduo@mario]build_gunrock$ bin/connected_component market /data/gunrock_dataset/large/ak2010/ak2010.mtx
Using device 0: Tesla K40c
Reading from /data/gunrock_dataset/large/ak2010/ak2010.mtx:
  Parsing MARKET COO format (45292 nodes, 108549 directed edges)... Done parsing (0s).
  Converting 45292 vertices, 108549 directed edges (unordered tuples) to CSR format...
Done converting (0s).

Degree Histogram (45292 vertices, 108549 edges):
    Degree   0: 9827 (21.70%)
    Degree 2^0: 10029 (22.14%)
    Degree 2^1: 15747 (34.77%)
    Degree 2^2: 7953 (17.56%)
    Degree 2^3: 1407 (3.11%)
    Degree 2^4: 275 (0.61%)
    Degree 2^5: 41 (0.09%)
    Degree 2^6: 12 (0.03%)
    Degree 2^7: 1 (0.00%)

Number of Components: 50
Top 10 largest components:
CC ID: 0, CC Root: 0, CC Size: 42381
CC ID: 1, CC Root: 22027, CC Size: 1335
CC ID: 2, CC Root: 44, CC Size: 524
CC ID: 3, CC Root: 1105, CC Size: 446
CC ID: 4, CC Root: 6663, CC Size: 96
CC ID: 5, CC Root: 1113, CC Size: 70
CC ID: 6, CC Root: 1106, CC Size: 60
CC ID: 7, CC Root: 29021, CC Size: 51
CC ID: 8, CC Root: 21, CC Size: 45
CC ID: 9, CC Root: 1125, CC Size: 34
[GPU Connected Component] finished.
 elapsed: 1.8894 ms
 num_iterations: 7

**Comment by ydwu** [comment]
fixed in 1d2ff81fdb1515b44bf7033b7076e2a32f33ebf7.
